src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilder.cs

```diff
@@ -181,7 +179,8 @@ public ClusterBuilder RegisterStreamFactory(Func<IStreamFactory, IStreamFactory>
         {
             Ensure.IsNotNull(wrapper, nameof(wrapper));
 
-            _streamFactoryWrapper = inner => wrapper(_streamFactoryWrapper(inner));
+            var previous = _streamFactoryWrapper; // use a local variable to ensure the previous values is captured properly by the lambda
```

Comment:
I used "previous values" instead of "previous value" by mistake in my comment. Sorry.

Might as well fix the comment to be grammatical.